### PR TITLE
Update Wasm VM code to the latest from envoy-wasm.

### PR DESCRIPTION
**Pull request by jplevyak**
- make v8 cloneable      
- add VM stats and debug logging. 
- make v8 less build environmentally sensitive/hermetic.

### CODE REVIEW

**Comment by jplevyak** [comment]
@lizan

File: bazel/external/wee8.genrule_cmd [hunk 1]
```diff
@@ -71,7 +71,11 @@ WEE8_BUILD_ARGS+=" v8_use_external_startup_data=false"
 WEE8_BUILD_ARGS+=" v8_enable_shared_ro_heap=false"
 
 # Build wee8.
-third_party/depot_tools/gn gen out/wee8 --args="$$WEE8_BUILD_ARGS"
+if [[ `uname` == "Darwin" ]]; then
```
[the remark this answers is not in the document]
Comment (PiotrSikora) [reply bazel/external/wee8.genrule_cmd]: It is, but for some reason it doesn't work on @jplevyak's box.

File: source/extensions/common/wasm/wasm_vm.cc [hunk 2]
```diff
@@ -3,6 +3,7 @@
 #include <memory>
 
 #include "extensions/common/wasm/null/null.h"
+
```
Comment (PiotrSikora) [inline source/extensions/common/wasm/wasm_vm.cc]: Nit: unnecessary empty line.

File: bazel/external/wee8.genrule_cmd (hunk 1 above)
[the remark this answers is not in the document]
Comment (jplevyak) [reply bazel/external/wee8.genrule_cmd]: That script also gropes around in the environment in a very non-hygienic way in order to do this very simple thing, which is probably why it doesn't work on my dev box.

File: source/extensions/common/wasm/wasm_vm.cc (hunk 2 above)
Comment (jplevyak) [reply source/extensions/common/wasm/wasm_vm.cc]: Fixed.

File: source/extensions/common/wasm/null/null_vm.cc [hunk 3]
```diff
@@ -17,6 +17,8 @@ namespace Common {
 namespace Wasm {
 namespace Null {
 
+VmGlobalStats global_stats_;
```
[the remark this answers is not in the document]
Comment (jplevyak) [reply source/extensions/common/wasm/null/null_vm.cc]: Done.

File: source/extensions/common/wasm/BUILD [hunk 4]
```diff
@@ -25,6 +25,14 @@ envoy_cc_library(
     ],
 )
 
+envoy_cc_library(
+    name = "wasm_vm_base",
+    hdrs = ["wasm_vm_base.h"],
```
[the remark this answers is not in the document]
Comment (jplevyak) [reply source/extensions/common/wasm/BUILD]: Woops.  Done.

**Comment by PiotrSikora** [comment]
Not stale? @lizan @jplevyak

**Comment by jplevyak** [comment]
This has been replaced with https://github.com/envoyproxy/envoy/pull/9259

**Pull request closed without merging**